src/Viewer/services/utils.js

```diff
@@ -44,27 +44,42 @@ const modifyPage = (action, currentPage, requestedPage, pages) => {
 };
 
 /**
- * Updates the URL based on the provided file metadata. If file
- * was downloaded using filePath, indicate the location in URL.
+ * Get modified URL from `window.location` based on the provided search and
+ * hash parameters.
  *
- * @param {object} fileMetadata
- * @param {number} logEventIdx
+ * @param {object} searchParams
+ * @param {object} hashParams
+ * @return {string} modified URL
  */
-const modifyFileMetadata = (fileMetadata, logEventIdx) => {
-    if (fileMetadata && fileMetadata.filePath) {
-        let url = `${window.location.origin}${window.location.pathname}`;
-        url += `?filePath=${fileMetadata.filePath}`;
-        if (0 !== logEventIdx) {
-            url += `#logEventIdx=${logEventIdx}`;
+const getModifiedUrl = (searchParams, hashParams) => {
+    const url = new URL(`${window.location.origin}${window.location.pathname}`);
+
+    const urlSearchParams = new URLSearchParams(window.location.search.substring(1));
+    const urlHashParams = new URLSearchParams(window.location.hash.substring(1));
+
+    Object.entries(searchParams).forEach(([key, value]) => {
+        urlSearchParams.delete(key);
+        if (null !== value) {
+            urlSearchParams.set(key, value.toString());
         }
-        window.history.pushState({path: url}, "", url);
-    } else if (fileMetadata && !fileMetadata.filePath) {
-        let url = `${window.location.origin}${window.location.pathname}`;
-        if (0 !== logEventIdx) {
-            url += `#logEventIdx=${logEventIdx}`;
+    });
+    Object.entries(hashParams).forEach(([key, value]) => {
+        urlHashParams.delete(key);
+        if (null !== value) {
+            urlHashParams.set(key, value.toString());
         }
-        window.history.pushState({path: url}, "", url);
+    });
+
+    let urlSearchParamsAsString = urlSearchParams.toString();
+    if (false === /%3F|%23|%26/.test(urlSearchParamsAsString)) {
+        // avoid encoding the URL if it does not contain reserved characters
+        urlSearchParamsAsString = decodeURIComponent(urlSearchParamsAsString);
     }
+
+    url.search = urlSearchParamsAsString;
+    url.hash = decodeURIComponent(urlHashParams.toString());
```

Comment:
Initial thought was to avoid encoding harmless characters (see above discussion) but there has not been any real use case for such. Will remove in the next commit.